### PR TITLE
Modernize vendor code

### PR DESCRIPTION
**Pull request by locks**
Fixes https://github.com/emberjs/ember-2-legacy/issues/44.

### CODE REVIEW

**Comment by rwjblue** [comment]
https://github.com/emberjs/ember-2-legacy/pull/47 should resolve the CI issues here...

**Comment by locks** [comment]
We are green :)

File: vendor/component-defaultlayout.js [hunk 1]
```diff
@@ -1,5 +1,5 @@
 (function() {
```
Comment (thoov) [inline vendor/component-defaultlayout.js]: we can convert:

```js
(function() {
  // ...
})();
```
over to:

```js
(() => {
  // ...
})();
```

File: vendor/router-resource.js [hunk 2]
```diff
@@ -1,5 +1,5 @@
 (function() {
```
Comment (thoov) [inline vendor/router-resource.js]: I had to change the below resource function from https://github.com/emberjs/ember.js/pull/15883/files#diff-57d1dbf7bb82d085435a5e830a2e77c7. We should move back to the original default arguments

**Comment by locks** [comment]
I'm gonna say this is not valuable anymore ;P Thanks for the review peeps!

**Pull request closed without merging**